piff/util.py

```diff
@@ -195,6 +195,28 @@ def hsm(star):
 
     return flux, center.x, center.y, sigma, shape.g1, shape.g2, flag
 
+def estimate_cov_from_jac(jac):
+    """Estimate a covariance matrix from a jacobian as returned by scipy.optimize.least_squares
+    .. math::
+        C = (J^T J)^{-1}
+    This is computed using Moore-Penrose inversion to discard singular values.
+
+    :param jac:     The Jacobian as a 2d numpy array
+
+    :returns: cov, a numpy array giving the estimated covariance.
+    """
+    import scipy
+    # Cribbed from implemenation in scipy.optimize.curve_fit
+    # https://github.com/scipy/scipy/blob/maintenance/1.3.x/scipy/optimize/minpack.py#L771
+
+    # Do Moore-Penrose inverse discarding zero singular values.
+    _, s, VT = scipy.linalg.svd(jac, full_matrices=False)
```

Comment:
Is it safer to do a try/except for the SVD decomposition and return a +infinite in the variance in the same way it is done in pixelgrid?

Reply:
I don't think the svd can fail, at least for the same kinds of reasons that the `inv` usage in PixelGrid can fail (e.g. for a singular matrix).  But just in case, I added a try/except block here.  There might be some kind of input matrix that could cause problems, so better safe than sorry.